torchao/prototype/safetensors/safetensors_serialization.py

```diff
@@ -0,0 +1,160 @@
+import dataclasses
+import enum
+import json
+from typing import Any, Dict
+
+import torch
+
+import torchao
+from torchao.quantization import Float8Tensor
+from torchao.quantization.quantize_.common import KernelPreference
+from torchao.quantization.quantize_.workflows import QuantizeTensorToFloat8Kwargs
+
+ALLOWED_CLASSES = {
+    "Float8Tensor": torchao.quantization.Float8Tensor,
+    "Float8MMConfig": torchao.float8.inference.Float8MMConfig,
+    "QuantizeTensorToFloat8Kwargs": QuantizeTensorToFloat8Kwargs,
+    "PerRow": torchao.quantization.granularity.PerRow,
+    "KernelPreference": KernelPreference,
+}
+
+
+class Float8TensorAttributeJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, Float8Tensor):
+            tensor_attr_dict = {}
+            all_tensor_attributes = (
+                o.optional_tensor_attribute_names + o.tensor_attribute_names
+            )
+
+            for tensor_attribute_name in all_tensor_attributes:
+                attribute = getattr(o, tensor_attribute_name)
+                serialized_attribute = self.encode_value(attribute)
```

Comment:
nit: might be better to say `encoded_attribute` I think, serialization is an overloaded term